x-pack/plugins/search_inference_endpoints/public/components/all_inference_endpoints/render_table_columns/render_actions/use_actions.tsx

```diff
@@ -0,0 +1,107 @@
+  EuiTableComputedColumnType,
+  EuiContextMenuPanelDescriptor,
+  EuiContextMenuPanelItemDescriptor,
+} from '@elastic/eui';
+import { EuiButtonIcon, EuiContextMenu, EuiPopover } from '@elastic/eui';
+import React, { useCallback, useMemo, useState } from 'react';
+import { InferenceEndpointUI } from '../../types';
+import { useCopyIDAction } from './actions/copy_id/use_copy_id_action';
+import { useDeleteAction } from './actions/delete/use_delete_action';
+import { ConfirmDeleteEndpointModal } from './actions/delete/confirm_delete_endpoint';
+
+const ActionColumnComponent: React.FC<{ interfaceEndpoint: InferenceEndpointUI }> = ({
+  interfaceEndpoint,
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const tooglePopover = useCallback(() => setIsPopoverOpen(!isPopoverOpen), [isPopoverOpen]);
+  const closePopover = useCallback(() => setIsPopoverOpen(false), []);
+  const refreshCases = () => {};
+
+  const copyIDAction = useCopyIDAction({
+    onActionSuccess: closePopover,
+  });
+
+  const deleteAction = useDeleteAction({
+    onAction: closePopover,
+    onActionSuccess: refreshCases,
+  });
+
+  const panels = useMemo((): EuiContextMenuPanelDescriptor[] => {
+    const mainPanelItems: EuiContextMenuPanelItemDescriptor[] = [];
+    const panelsToBuild: EuiContextMenuPanelDescriptor[] = [{ id: 0, items: mainPanelItems }];
+
+    mainPanelItems.push(copyIDAction.getAction(interfaceEndpoint));
+    mainPanelItems.push(deleteAction.getAction(interfaceEndpoint));
+
+    return panelsToBuild;
+  }, [copyIDAction, deleteAction, interfaceEndpoint]);
+
+  return (
+    <>
+      <EuiPopover
+        id={`inference-action-popover-${interfaceEndpoint.endpoint}`}
+        key={`inference-action-popover-${interfaceEndpoint.endpoint}`}
+        button={
+          <EuiButtonIcon
+            onClick={tooglePopover}
+            iconType="boxesHorizontal"
+            aria-label={'Actions'}
+            color="text"
+            key={`inference-action-popover-button-${interfaceEndpoint.endpoint}`}
+            disabled={false}
+          />
+        }
+        isOpen={isPopoverOpen}
+        closePopover={closePopover}
+        panelPaddingSize="none"
+        anchorPosition="downLeft"
+      >
+        <EuiContextMenu
+          initialPanelId={0}
+          panels={panels}
+          key={`inference-action-menu-${interfaceEndpoint.endpoint}`}
+        />
+      </EuiPopover>
+      {deleteAction.isModalVisible ? (
+        <ConfirmDeleteEndpointModal
+          onCancel={deleteAction.onCloseModal}
+          onConfirm={deleteAction.onConfirmDeletion}
+        />
+      ) : null}
+    </>
+  );
+};
+
+ActionColumnComponent.displayName = 'ActionColumnComponent';
+
+const ActionColumn = React.memo(ActionColumnComponent);
+
+interface UseBulkActionsReturnValue {
+  actions: EuiTableComputedColumnType<InferenceEndpointUI>;
+}
+
+export const useActions = (): UseBulkActionsReturnValue => {
+  return {
+    actions: {
+      name: '',
```

Comment:
should this have a `name`?

Reply:
According to the [design](https://www.figma.com/design/duLXTmppjDWJyufbr5jyLW/Inference-Endpoint-Management-[Search-170]?node-id=2936-226089&t=ReXt6PZL7l0cwZ6e-0), there should be no name for ActionColumn.

cc: @mdefazio